### DISCUSSION TITLE
Insert custom meta tags in file

### DISCUSSION THREAD

**Discussion by rodrigolki** [dompdf/dompdf#2447] [Q&A]
Hello everyone!

Is there some way to add custom meta tags on PDF file? Not common tags like author, title, description i need to insert some tags from certificate validation. 

Someone faced this problem and could help?

**Comment by bsweeney** [comment]
I'm a little rusty on PDF metadata, do you have a sample of what you want to add? And is this using the CPDF back end?

For the standard info element CPDF is limited to the [following properties](https://github.com/dompdf/dompdf/blob/v1.0.2/lib/Cpdf.php#L1750-L1758):
- Title
- Author
- Subject
- Keywords
- Creator
- Producer
- CreationDate
- ModDate
- Trapped

The library does not yet support other types of metadata, such as XMP.

**Reply by rodrigolki** [reply]
The custom metadata are for medical signed documents, we look into CPDF library and find a way to manipulate it to accept any kind of metadata, and this seems to atend our goal, the only issue its that we need to download the lib and save into our repo, removing it from composer.

In below its a print of the metadata that we need.

![image](https://user-images.githubusercontent.com/8506469/116913860-60544b00-ac20-11eb-96c5-5b175d1cda3d.png)

**Reply by bsweeney** [reply]
OK that's what I was thinking. Adding support to CPDF for custom properties in the document information dictionary should be straightforward. I suspect PDFLib already supports it but would have to confirm.

A bit more thought would have to be put forward to support using metadata from the HTML document. For example, we should almost certainly exclude http-equiv meta tags, but there are also a lot of psuedo-standard name values that would not really make sense as PDF metadata (see the list here: https://gist.github.com/lancejpollard/1978404).

I created an enhancement request for this feature in #2448. You are welcome to submit a pull request to accommodate that enhancement.
